File: 0100-same-tree/0100-same-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        var stack = new Stack<(TreeNode, TreeNode)>();
        stack.Push((p, q));

        while (stack.Count > 0) {
            var (pNode, qNode) = stack.Pop();
            if (pNode == null && qNode== null) continue;
            if (pNode?.val != qNode?.val) return false;
            stack.Push((pNode.right, qNode.right));
            stack.Push((pNode.left, qNode.left));
        }

        return true;
    }
}

File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        stack_p = []
        stack_q = []
        
        stack_p.append(p)
        stack_q.append(q)
        
        while stack_p and stack_q:
            node_p, node_q = stack_p.pop(), stack_q.pop()
            if not node_p and not node_q: continue
            if not node_p or not node_q or node_p.val != node_q.val:
                return False
            stack_p.append(node_p.right)
            stack_p.append(node_p.left)
            stack_q.append(node_q.right)
            stack_q.append(node_q.left)
            
        if stack_p or stack_q:
            return False
        
        return True
